docker/test_env/ethereum_env.go

```diff
@@ -0,0 +1,442 @@
+package test_env
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/rs/zerolog"
+	tc "github.com/testcontainers/testcontainers-go"
+	tcwait "github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/blockchain"
+	"github.com/smartcontractkit/chainlink-testing-framework/docker"
+	"github.com/smartcontractkit/chainlink-testing-framework/logging"
+)
+
+type ConsensusType string
+
+const (
+	ConsensusType_PoS ConsensusType = "pos"
+	ConsensusType_PoW ConsensusType = "pow"
+)
+
+type ExecutionLayer string
+
+const (
+	ExecutionLayer_Geth ExecutionLayer = "geth"
+)
+
+type ConsensusLayer string
+
+const (
+	ConsensusLayer_Prysm ConsensusLayer = "prysm"
+)
+
+type EthereumNetworkBuilder struct {
+	t                 *testing.T
+	l                 zerolog.Logger
+	dockerNetworks    []string
+	consensusType     *ConsensusType
+	consensusLayer    *ConsensusLayer
+	consensusNodes    int
+	executionLayer    *ExecutionLayer
+	executionNodes    int
+	beaconChainConfig *BeaconChainConfig
+	validated         bool
+	existingConfig    *EthereumNetworkConfig
+}
+
+type Eth2Components struct {
+	Geth        *Geth2
+	BeaconChain *PrysmBeaconChain
+	Validator   *PrysmValidator
+}
+
+func NewEthereumNetworkBuilder(t *testing.T) EthereumNetworkBuilder {
```

Comment:
@Tofel can we decouple it from `testing.T`? This way we can use it outside of `go test`. For example in a CLI to start test env

Reply:
we could, but then logger won't be associated to any test, but what impact would that have?

Reply:
@Tofel hm, if `testing.T` is only needed for a logger then maybe have `WithLogger()` that takes a logger that already has `testing.T`?

Reply:
okay, I found a way that should work